### PR TITLE
feat: add custom args

### PR DESCRIPTION
**Pull request by goldenxinxing**
## Description

1. write arguments spec to job.yaml
2. support parse arguments in server side
3. output the failed log in githua action

## Modules
- [ ] UI
- [x] Controller
- [x] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [x] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: client/starwhale/api/_impl/argument.py [hunk 1]
```diff
@@ -94,14 +65,25 @@ def asdict(self) -> t.Dict[str, t.Any]:
...
-                    field = OptionField(**option.to_info_dict())
+                    field = OptionFieldClient(**option.to_info_dict())
                     if PYDANTIC_V2:
                         info = field.model_dump(mode="json")
                     else:
                         info = field.dict()
                     r[func][dtype][option.name] = info
         return r
 
+    def asobj(self) -> t.Dict[str, t.Dict[str, t.Dict[str, OptionField]]]:
```
Comment (tianweidut) [inline client/starwhale/api/_impl/argument.py:76]: define a type alias for `t.Dict[str, t.Dict[str, OptionField]]`?  StepSpec.arguments field also needs this type.

File: client/starwhale/base/client/models/models.py [hunk 2]
```diff
@@ -1480,6 +1480,26 @@ class ServiceSpec(SwBaseModel):
...
+    name: str
+    param_type: str
+    # for Choice type
+    choices: Optional[List[str]] = None
+    case_sensitive: bool = False
+
+
+class OptionField(SwBaseModel):
+    name: str
+    opts: List[str]
```
Comment (tianweidut) [inline client/starwhale/base/client/models/models.py:1493]: Maybe we can remove `opts` field, only use `name` field as input argument name.

**Comment by tianweidut** [comment]
Others LGTM

File: client/starwhale/base/client/models/models.py (hunk 2 above)
Comment (goldenxinxing) [reply client/starwhale/base/client/models/models.py]: removed

File: client/starwhale/api/_impl/argument.py (hunk 1 above)
Comment (goldenxinxing) [reply client/starwhale/api/_impl/argument.py]: done，but StepSpec.arguments was generated by pydantic, so just used in argument.py

**Review by tianweidut — APPROVED** [review]

**Pull request merged**